packages/webdriverio/src/commands/browser/mock.ts

```diff
@@ -172,6 +177,25 @@ export async function mock (
     const networkInterception = new NetworkInterception(url, filterOptions, browser)
     SESSION_MOCKS[handle].add(networkInterception as Interception)
 
+    networkInterception.remove = () => {
```

Comment:
I agree, this is not the right place to define this command in and we should find ways to put it in `NetworkInterception`.

> you need networkInterception in order to remove it from set

If you define this method in the `NetworkInterception` class it is accessible through `this`.

> you need SESSION_MOCKS, to know, where to delete from

`SESSION_MOCKS` then needs to be defined somewhere where `mock.ts` and the file where `NetworkInterception` is defined can import it from without having circular dependencies.

> you need handle, to know where to exactly delete from

make `remove` async and get it via `const handle = await browser.getWindowHandle()`

> you need this client: CDPSession to disable fetch domain,

You have access to the `browser` variable so you can get that variable too.

Reply:
The method `mockRestore` is meant to do what you want to do with `remove` .. can we just make sure the restore method ensures that nor memory is hogged?

Reply:
> This is an issue that should be addressed but I am not happy with the current approach

That's why i converted it to draft: i found a better way to do it

Reply:
> You have access to the browser variable so you can get that variable too

i tried, and that's what i discovered: you can get puppeteer through browser, but not `client: CDPSession`, because it would be different client, and we need original one.

Reply:
> The method `mockRestore` is meant to do what you want to do with `remove` .. can we just make sure the restore method ensures that nor memory is hogged?

There is also a moment to consider. There are two options:
- `restore` also sends `Fetch.disable`, if there are no other mocks (if `restore` means "i dont need this anymore". In this case we would have to enable fetch domain again after some `respond` / `abort`)
- `restore` does not disable fetch domain, even if there are no other mocks. (if you plan to use `respond` / `abort` after it was restored. In this case we could just ensure this mock presents in `SESSION_MOCKS`)

First option also makes `restore` async for  devtools and breaks existing "respond / abort after restore" cases, so i think about implementing second approach, but at first i need to clarify: what happens if you try to "respond / abort after restore" with saucelabs? Will it work just like mock wasn't restored?

Reply:
We have designed the API to be as close as possible to the [Jest Mock API](https://jestjs.io/docs/mock-function-api#mockfnmockrestore) so I would say the first should be more appropriate. I feel honestly less worried about the Sauce Labs integration. I was working there at a time I build this and don't have any usage numbers. At the end of the day this API should support what will be supported in WebDriver Bidi because Sauce eventually has to support that.